util/soc_codes_names.py

```diff
@@ -0,0 +1,2985 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""SOCv2018 codes -> Occupation names"""
+
+import re
+import requests
+
+
+def _create_soc_codes_map():
+    """Generates SOC codes to occupation map from the SOCv2018 MCF file"""
+
+    # A regex to clean SOC names. This regex catches non alphanumeric characters
+    # (excluding whitespace), 'and', 'Occupations'.
+    soc_name_clean_regex = re.compile(r'[^A-Za-z0-9 ]+|\band\b|\bOccupations\b')
+
+    soc_map = dict()
+    soc_mcf = requests.get(
+        'https://raw.githubusercontent.com/datacommonsorg/schema/main/core/soc.mcf'
+    ).text
+    soc_mcf = soc_mcf.split('\n')
+
+    for line in soc_mcf:
+        # Extracting the SOC code from the dcid
+        # The dcid is of the form 'dcid:SOCv2018/<code>'
+        if line.startswith('Node:'):
+            if 'SOCv2018' in line:
```

Comment:
if 'SOCv2018/' in line:
  so split()[1] is always present

Reply:
Done.